options.lua

```diff
@@ -0,0 +1,69 @@
+local profile = Game.GetProfile()
+
+local function split_string(inputstr, sep)
+    if sep == nil then
+        sep = "%s"
+    end
+    local t = {}
+    for str in string.gmatch(inputstr, "([^" .. sep .. "]+)") do
+        table.insert(t, str)
+    end
+    return t
+end
+
+local function update_gui(menu, metric, value)
+    menu[metric].value = value
+    menu[metric .. '_text'].text = string.format("%.0f%%", value)
+end
+
+local Mod_layout =
+[[
+<VerticalList child_padding=8>
+    <Text fill=false text="Health" style=hl textalign=left />
+    <HorizontalList child_fill=true>
+        <Text text="0" fill=false valign=center textalign=right width=50 margin_right=5 id=health_text/>
+        <Slider height=36 min=0 max=100 step=1 id=health on_change={on_slider_changed}/>
+    </HorizontalList>
+
+    <Text fill=false text="Damage" style=hl textalign=left />
+    <HorizontalList child_fill=true>
+        <Text text="0" fill=false valign=center textalign=right width=50 margin_right=5 id=damage_text/>
+        <Slider height=36 min=0 max=100 id=damage on_change={on_slider_changed}/>
+    </HorizontalList>
+</VerticalList>
+]]
+
+return UI.New(Mod_layout, {
```

Comment:
The options GUI only updates the profile, but the profile is only used on init meaning that a user would have to reload in order for it to take effect. I think if you add `destruct` and make the function do the same updating of the frames (preferably sharing the logic with init in a separate function) I think that might be a better user experience.

Thoughts?

Reply:
Mmm I realized the game might get angry about this.  If it does maybe just adding a note to reload or something is worth it?